### ISSUE TITLE
[Bug]: [master & 2.2.0] metric type check will be ignored if all entities are filtered out

### ISSUE THREAD

**Issue by longjiquan** [milvus-io/milvus#24555]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

https://github.com/milvus-io/milvus/blob/5358da2a10820cc685587a56936b0ec65ffdec64/internal/core/src/query/visitors/ExecPlanNodeVisitor.cpp#L94-L98

**Comment by longjiquan** [comment]
the root cause of https://github.com/milvus-io/milvus/issues/24533

**Comment by longjiquan** [comment]
won't be fixed in 2.2.9, we'll solve this in the later version.

**Comment by bigsheeper** [comment]
fixed
